source/MainPanel.cpp

```diff
@@ -110,6 +110,9 @@ void MainPanel::Step()
 			isActive = !DoHelp("friendly disabled");
 		if(isActive && player.Ships().size() > 1)
 			isActive = !DoHelp("multiple ship controls");
+		double scanRange = 100. * sqrt(flagship->Attributes().Get("asteroid scan power"));
```

Comment:
Why the math?

Reply:
I was "monkey see monkey do" adding this code.

In other parts of the code I saw "valid asteroid targeting" as a math equation.  Is simply checking for the attribute good enough?  Is it ever possible for `flagship->Attributes().Get("asteroid scan power")` to be not a `nullptr` and return `0.`?

I would have originally done the following:

```cpp
if(isActive && flagship->IsTargetable() && flagship->Attributes().Get("asteroid scan power") && player.Ships().size() > 1)
```

But I was afraid of edge case where `flagship->Attributes().Get("asteroid scan power") == 0.`  Is `(double) 0.` falsey in logic?  I tried looking it up but struggled to find a truth table that made sense to me.

Reply:
`0` (integral) and `0.` (floating point) are both equivalent to false in a boolean context.

If the player has _any_ asteroid scanning power, then they should get the tutorial.

Reply:
I will make this change.